Build failure with ghc 8.8

ghc 8.8 & its corresponding `base` have finally finished extracting `fail` from `Monad`, which has resulted in a build failure:

```
Opentype/Fileformat/Name.hs:87:24: error:
    • No instance for (MonadFail (Either String))
        arising from a use of ‘fail’
    • In the second argument of ‘($)’, namely
        ‘fail "Unsupported name table format."’
      In a stmt of a 'do' block:
        when (version > 0) $ fail "Unsupported name table format."
      In the expression:
        do version <- index16 bs 0
           when (version > 0) $ fail "Unsupported name table format."
           len <- index16 bs 1
           storage <- index16 bs 2
           ....
   |
87 |   when (version > 0) $ fail "Unsupported name table format."
   |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

--- a/Opentype/Fileformat/Cmap.hs
+++ b/Opentype/Fileformat/Cmap.hs
@@ -257,7 +257,7 @@ putMap2 cmap = do
                      (fromIntegral hb `shift` 8 .|. 0xff) $
                      glyphMap cmap
             (fstCode, lstCode)
-              | M.null subMap = (0, -1)
+              | M.null subMap = (0, maxBound)
               | otherwise = (fst $ M.findMin subMap,
                              fst $ M.findMax subMap)
             ec = lstCode - fstCode + 1

--- a/Opentype/Fileformat/Cmap.hs
+++ b/Opentype/Fileformat/Cmap.hs
@@ -3,7 +3,6 @@ import Opentype.Fileformat.Types
 import Data.Binary
 import Data.Binary.Put
 import Data.List (sort, mapAccumL, foldl')
-import Data.Either (either)
 import Control.Monad
 import Data.Traversable (for)
 import Data.Foldable (for_, traverse_)
@@ -135,7 +134,7 @@ readCmapTable :: Strict.ByteString -> Either String CmapTable
 readCmapTable bs = do
   version <- index16 bs 0
   when (version /= 0) $
-    fail "unsupported cmap version."
+    Left "unsupported cmap version."
   n <- index16 bs 1
   entries <- for [0..n-1] $ \i -> do
     pfID <- toPf =<< (index16 bs $ 2 + i*4)
@@ -166,16 +165,15 @@ readCmap bs_ = do
   c <- index16 bs_ 0
   let bs | (c >= 8 && c < 14) = Strict.drop 8 bs_
          | otherwise = Strict.drop 4 bs_
-  either fail return $
-    case c of 
-      0 -> getMap0 bs
-      2 -> getMap2 bs
-      4 -> getMap4 bs
-      6 -> getMap6 bs
-      8 -> getMap8 bs
-      10 -> getMap10 bs
-      12 -> getMap12 bs
-      i -> fail $ "unsupported map encoding " ++ show i
+  case c of
+    0 -> getMap0 bs
+    2 -> getMap2 bs
+    4 -> getMap4 bs
+    6 -> getMap6 bs
+    8 -> getMap8 bs
+    10 -> getMap10 bs
+    12 -> getMap12 bs
+    i -> Left $ "unsupported map encoding " ++ show i
 
 subIntMap :: Word32 -> Word32 -> WordMap GlyphID -> WordMap GlyphID
 subIntMap from to intMap =

--- a/Opentype/Fileformat/Name.hs
+++ b/Opentype/Fileformat/Name.hs
@@ -84,7 +84,7 @@ putNameTable (NameTable records_) = do
 readNameTable :: Strict.ByteString -> Either String NameTable
 readNameTable bs = do
   version <- index16 bs 0
-  when (version > 0) $ fail "Unsupported name table format."
+  when (version > 0) $ Left "Unsupported name table format."
   len <- index16 bs 1 
   storage <- index16 bs 2
   records <- for [0..len-1] $ \i -> do

